iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/patent/PatentMetadataRetrieverJob.java

```diff
@@ -0,0 +1,167 @@
+        jcommander.parse(args);
+        
+        OutputPaths outputPaths = new OutputPaths(params);
+        
+        try (JavaSparkContext sc = JavaSparkContextFactory.withConfAndKryo(new SparkConf())) {
+            HdfsUtils.remove(sc.hadoopConfiguration(), params.outputPath);
+            HdfsUtils.remove(sc.hadoopConfiguration(), params.outputFaultPath);
+            HdfsUtils.remove(sc.hadoopConfiguration(), params.outputReportPath);
+            
+            try {
+                PatentServiceFacade patentServiceFacade = ServiceFacadeUtils.instantiate(sc.hadoopConfiguration());
+                
+                JavaRDD<ImportedPatent> importedPatents = avroLoader.loadJavaRDD(sc, params.inputPath,
+                        ImportedPatent.class);
+
+                Tuple2<JavaRDD<DocumentText>, JavaRDD<Fault>> results = obtainMetadata(importedPatents, patentServiceFacade);
+                
+                storeInOutput(results._1, results._2, generateReportEntries(sc, results._1, results._2), outputPaths,
+                        params.numberOfEmittedFiles);
+                
+            } catch (ServiceFacadeException e) {
+                throw new RuntimeException("unable to instantiate patent service facade!", e);
+            }
+        }
+    }
+
+    //------------------------ PRIVATE --------------------------
+
+    private static JavaRDD<ReportEntry> generateReportEntries(JavaSparkContext sparkContext, 
+            JavaRDD<DocumentText> processedEntities, JavaRDD<Fault> processedFaults) {
+        
+        Preconditions.checkNotNull(sparkContext, "sparkContext has not been set");
+        
+        ReportEntry processedEnttiesCounter = ReportEntryFactory.createCounterReportEntry(COUNTER_PROCESSED_TOTAL, processedEntities.count());
+        ReportEntry processedFaultsCounter = ReportEntryFactory.createCounterReportEntry(COUNTER_PROCESSED_FAULT, processedFaults.count());
+        
+        return sparkContext.parallelize(Lists.newArrayList(processedEnttiesCounter, processedFaultsCounter));
+    }
+    
+    private static void storeInOutput(JavaRDD<DocumentText> retrievedPatentMeta, 
+            JavaRDD<Fault> faults, JavaRDD<ReportEntry> reports, OutputPaths outputPaths, int numberOfEmittedFiles) {
+        avroSaver.saveJavaRDD(retrievedPatentMeta.coalesce(numberOfEmittedFiles), DocumentText.SCHEMA$, outputPaths.getResult());
+        avroSaver.saveJavaRDD(faults.coalesce(numberOfEmittedFiles), Fault.SCHEMA$, outputPaths.getFault());
+        avroSaver.saveJavaRDD(reports.coalesce(numberOfEmittedFiles), ReportEntry.SCHEMA$, outputPaths.getReport());
+    }
+    
+    private static Tuple2<JavaRDD<DocumentText>, JavaRDD<Fault>> obtainMetadata(JavaRDD<ImportedPatent> input, PatentServiceFacade patentServiceFacade) {
+        JavaPairRDD<CharSequence, ContentRetrieverResponse> obtainedIdToResponce = input
+                .mapToPair(x -> new Tuple2<CharSequence, ContentRetrieverResponse>(getId(x),
```

Comment:
I think this one is still not changed.

Reply:
This is quite strange, because I remember fixing this issue. Anyway I have fixed that again. Probably I've been withdrawing changes and used `ctrl+z` too many times...